cmd/minikube/cmd/start_flags.go

```diff
@@ -286,12 +288,17 @@ func generateClusterConfig(cmd *cobra.Command, existing *config.ClusterConfig, k
 			out.WarningT("With --network-plugin=cni, you will need to provide your own CNI. See --cni flag as a user-friendly alternative")
 		}
 
+		if !driver.IsKIC(drvName) && viper.GetString(containerNetwork) != "" {
+			exit.Advice(reason.Usage, "--network flag is only valid with the docker driver", "remove the --network flag from `minikube start`")
```

Comment:
I am thinking how about we only warn the user and not exit ? like if they have minikube on auto select driver,
but they want to use this driver if Minikube chooses docker....

I think we should change to out.Warrning

Reply:
Personally I think it makes sense to exit since it forces correct usage of the flag, and minimizes potential confusion.

Reply:
I agree -- we should not ignore the user's request.